src/main.rs

```diff
@@ -144,10 +144,16 @@ fn write_to_file(file: impl AsRef<Path>) {
             return;
         }
     }
-    generate_cargo_nix(fs::File::create(&file).expect(&format!(
-        "could not open file for writing: {}",
-        file.display()
-    )));
+
+    let mut temp_file = tempfile::Builder::new()
+        .tempfile()
+        .expect("could not create new temporary file");
+
+    generate_cargo_nix(&mut temp_file);
+
+    temp_file
+        .persist(path)
+        .unwrap_or_else(|_| panic!("could not write file to final location: {}", path.display()));
```

Comment:
Regarding the `_` thingy: maybe we should print the error instead of ignoring it?

Reply:
I decided not to print the error here since we had ignored it in the original code with `expect()` and I wanted to do the same. If you'd like for me to print the error, I'd be happy to add another commit!

BTW, I chose to use `unwrap_or_else()` here instead of `expect()` to avoid calling `format!()` needlessly in the happy path. This is essentially equivalent to `expect("<message>")`.

Reply:
I think printing the error could be helpful if it fails at all. There doesn't seem to be any harm, at the least. :-)

And yes, I understood `unwrap_or_else()` vs `expect()` .. it is minor (and personal preference), though I personally prefer to use the latter because it is shorter (and performance is not required in such cases, or have negligible impact).

Reply:
No worries, I think I'll add a commit printing the error in this case. 😄

Reply:
Done! 👍